guide/language-concepts/normal-ordering.markdown

```diff
@@ -19,19 +19,67 @@ You can override normal ordering in exceptional circumstances by making a
 promise in a class context and defining that class based on the outcome of
 another promise, or using the `depends_on` promise attribute.
 
-### Agent normal ordering
+## Agent normal ordering
 
 CFEngine tries to keep variable and class promises before starting to consider
-any other kind of promise. In this way, global variable and classes can be set.
+any other kind of promise. In this way, global variables and classes can be set.
 
 If you set variables based on classes that are determined by other variables,
 then you introduce an order dependence to the resolution that might be
```

Comment:
I don't understand "introduce an order dependance to the resolution that might be non-unique".